Modules/_sqlite/util.c

```diff
@@ -36,27 +36,18 @@ pysqlite_step(sqlite3_stmt *statement)
     return rc;
 }
 
-/**
- * Checks the SQLite error code and sets the appropriate DB-API exception.
- * Returns the error code (0 means no error occurred).
- */
-int
-_pysqlite_seterror(pysqlite_state *state, sqlite3 *db)
+static PyObject *
+get_exception_class(pysqlite_state *state, int errorcode)
```

Comment:
This is slighly confusing because normally returning `NULL` signifies an error. Could you please document this both on the call site and in this function? Otherwise the call site reads weirdly because 

```
    PyObject *exc_class = get_exception_class(state, errorcode);
    if (exc_class == NULL) {
        return errorcode;
    }
```

seems that is handling an error

Reply:
Yes, I see now that it can be a bit confusing. I'll add a comment. Thanks!

Reply:
Is it cleared with 3589a6a7163d0148bf0c524d0be3f56b563e9d19?

(FYI, I need to rebase onto `main` again bco. GH-26202)